tests/Validation/ValidationValidatorTest.php

```diff
@@ -5986,6 +5986,20 @@ public function testBeforeAndAfter()
 
         $v = new Validator($trans, ['x' => ['a' => ['v' => 'c']], 'y' => 'invalid'], ['x' => 'date', 'y' => 'date|before:x']);
         $this->assertTrue($v->fails());
+
+        $v = new Validator($trans, ['y' => '1970-01-01'], ['x' => 'nullable|date', 'y' => 'nullable|date|after:x']);
+        $this->assertTrue($v->passes());
```

Comment:
> but that test case is failing?

No, this is the test case that is failing (when `x` is missing from the array). This tests before and after the changes introduced by your PR.

Reply:
Ah, I found the issue, when compared against `null`, it's comparing against the current time (`now`):

This should properly capture the behaviour before and after the PR:

```php
        $v = new Validator($trans, ['y' => '1970-01-01'], ['x' => 'nullable|date', 'y' => 'nullable|date|before:x']);
        $this->assertTrue($v->passes());

        $v = new Validator($trans, ['y' => now()->addDay()], ['x' => 'nullable|date', 'y' => 'nullable|date|after:x']);
        $this->assertTrue($v->passes());

        $v = new Validator($trans, ['x' => null, 'y' => '1970-01-01'], ['x' => 'nullable|date', 'y' => 'nullable|date|before:x']);
        $this->assertTrue($v->passes());

        $v = new Validator($trans, ['x' => null, 'y' => now()->addDay()], ['x' => 'nullable|date', 'y' => 'nullable|date|after:x']);
        $this->assertTrue($v->passes());

```

Reply:
Is it a different bug? 
not a breaking change?

Reply:
not a breaking change

Reply:
lemme just remove that test case for now.

and solve that bug with separate PR?

Reply:
I don't see any bug...

Reply:
this test case was event failing before my PR
```php
$v = new Validator($trans, ['dates' => [['x' => '2024-02-02 12:00:00', 'y' => '2024-02-02 12:00:00']]], ['dates.*.x' => 'date', 'dates.*.y' => 'date|after:x']);
$this->assertTrue($v->passes());
```

but I just replace x with starts_at and y with ends_at this passes

```php
$v = new Validator($trans, ['dates' => [['start_at' => '2024-02-02 12:00:00', 'ends_at' => '2024-02-02 12:00:00']]], ['dates.*.start_at' => 'date', 'dates.*.ends_at' => 'date|after:start_at']);
$this->assertTrue($v->passes());
```